src/3d.js

```diff
@@ -326,19 +596,59 @@ canvas:focus { outline: none; }
 				return sprite.room === room.id;
 			});
 			[].concat(items, tiles, sprites).forEach(function (drawing) {
-				requestIdleCallback(function () {
-					var f = drawing.animation.frameIndex;
-					for (var i = 0; i < drawing.animation.frameCount; ++i) {
-						drawing.animation.frameIndex = i;
-						getTexture(drawing, room.pal);
+				// replace drawings marked with the #draw tag
+				if (hackOptions.replaceDrawingTag) {
+					var name = drawing.name || '';
+					var tag = name.match(/#draw[- _]?(TIL|SPR|ITM)[- _]?([a-zA-Z0-9]+)/);
+					if (tag) {
+						// console.log(tag);
+						var map;
+						// tag[1] is the first capturing group, it can be either TIL, SPR, or ITM
+						switch (tag[1]) {
+							case 'TIL':
+								map = bitsy.tile;
+								break;
+							case 'SPR':
+								map = bitsy.sprite;
+								break;
+							case 'ITM':
+								map = bitsy.item;
+								break;
+						}
+						// tag[2] is the second capturing group which returns drawing id
+						var id = tag[2];
+						drawing.drw = map[id].drw;
+						drawing.animation.frameCount = map[id].animation.frameCount;
+						drawing.animation.isAnimated = map[id].animation.isAnimated;
+						drawing.col = map[id].col;
 					}
-					drawing.animation.frameIndex = f;
-				})
+				}
+				// preload textures
+				// safari, edge and ie don't support requestIdleCallback. make sure to check
```

Comment:
oh good catch; would probably be better to add a shim for `requestIdleCallback` rather than doing nothing in this case though (the preload was a big improvement on subjective performance when i was initially working on this)

Reply:
sounds like it would require to use npm to find some suitable library like you did for polyfill that is being imported at the top? i'm not very good at npm yet... mb should try

Reply:
an npm module for it would be preferable yeah, but if you're not familiar with npm and would rather it not block this PR, we can leave this as-is for now and i can replace it with a module later (feel free to take the opportunity to learn some more about npm if you want though!)